File: app/src/main/java/com/prisyazhnuy/newsapp/data/local/preferences/PreferencesSource.java
package com.prisyazhnuy.newsapp.data.local.preferences;

/**
 * max.pr on 04.03.2018.
 */

public interface PreferencesSource {
    void setSort(int sortType);

    int getSort();

    void setCheckedSources(String sources);

    String getCheckedSources();

    void setDateFrom(String from);

    String getDateFrom();

    void setDateTo(String to);

    String getDateTo();
}

File: app/src/main/java/com/prisyazhnuy/newsapp/dataKotlin/preferences/PreferencesSource.kt
package com.prisyazhnuy.newsapp.dataKotlin.preferences

import android.arch.lifecycle.LiveData

/**
 * max.pr on 07.04.2018.
 */
interface PreferencesSource {

    fun setSort(type: String)

    fun getSort(): LiveData<String>
    fun getSource(): LiveData<String>
    fun getFrom(): LiveData<String>
    fun getTo(): LiveData<String>
    fun setSource(source: String)
    fun setFrom(from: String)
    fun setTo(to: String)


}